textgrad/engine/openai.py

```diff
@@ -96,3 +96,30 @@ def generate_with_messages(self, messages, temperature=0, max_tokens=2000, top_p
     def __call__(self, prompt, **kwargs):
         return self.generate(prompt, **kwargs)
 
+class AzureChatOpenAI(ChatOpenAI):
+    def __init__(
+        self,
+        model_string="gpt-35-turbo",
+        system_prompt=ChatOpenAI.DEFAULT_SYSTEM_PROMPT,
+        **kwargs):
+        """
+        :param model_string:
+        :param system_prompt:
+        """
+        root = platformdirs.user_cache_dir("textgrad")
+        cache_path = os.path.join(root, f"cache_azure_{model_string}.db")  # Changed cache path to differentiate from OpenAI cache
+
+        super().__init__(cache_path=cache_path, system_prompt=system_prompt, **kwargs)
+
+        self.system_prompt = system_prompt
+        api_version = os.getenv("AZURE_OPENAI_API_VERSION", "2023-07-01-preview")
```

Comment:
Maybe you don't want to hardcode `"2023-07-01-preview"` there but define as `FALLBACK_AZURE_OPENAI_API_VERSION` at the top or something.

Reply:
I have added 3 env, maybe 4 is too many.